src/openai/lib/azure.py

```diff
@@ -58,6 +58,58 @@ def _build_request(
                 options.url = f"/deployments/{model}{options.url}"
 
         return super()._build_request(options)
+    
+    def _configure_client_settings(
+        self,
+        api_key: str | None,
+        azure_ad_token: str | None,
+        azure_ad_token_provider: AzureADTokenProvider | AsyncAzureADTokenProvider | None,
+        azure_endpoint: str | None,
+        azure_deployment: str | None,
+        api_version: str | None,
+        base_url: str | None,
+        default_query: Mapping[str, object] | None
+    ) -> tuple[str, str | None, str, Mapping[str, object]]:
+        if api_key is None:
+            api_key = os.environ.get("AZURE_OPENAI_API_KEY")
+
+        if azure_ad_token is None:
+            azure_ad_token = os.environ.get("AZURE_OPENAI_AD_TOKEN")
+
+        if api_key is None and azure_ad_token is None and azure_ad_token_provider is None:
+            raise OpenAIError(
+                "Missing credentials. Please pass one of `api_key`, `azure_ad_token`, `azure_ad_token_provider`, or set the `AZURE_OPENAI_API_KEY` or `AZURE_OPENAI_AD_TOKEN` environment variables."
+            )
+
+        if api_version is None:
+            api_version = os.environ.get("OPENAI_API_VERSION", "v1")
```

Comment:
The original initialization code raises a ValueError if the api_version is still None after this step. I believe that should be preserved, otherwise `{"api-version": None}` could get added to `default_query`

Reply:
Ahh, no you still want to check the env var, just raise the error if api_verison isn't provided and there is no env var set. Here's the original init snippet:

```
if api_version is None:
    api_version = os.environ.get("OPENAI_API_VERSION"

if api_version is None:
    raise ValueError(
        "Must provide either the `api_version` argument or the `OPENAI_API_VERSION` environment variable"
    )
```